deployability/modules/testing/tests/test_central_components/test_stop.py

```diff
@@ -0,0 +1,102 @@
+
+import pytest
+
+from ..helpers.generic import HostInformation, GeneralComponentActions
+from ..helpers.manager import WazuhManager
+from ..helpers.dashboard import WazuhDashboard
+from ..helpers.indexer import WazuhIndexer
+from modules.testing.utils import logger
+
+
+@pytest.fixture(scope="module", autouse=True)
+def wazuh_params(request):
+    wazuh_version = request.config.getoption('--wazuh_version')
+    wazuh_revision = request.config.getoption('--wazuh_revision')
+    dependencies = request.config.getoption('--dependencies')
+    targets = request.config.getoption('--targets')
+
+    params = {
+        'wazuh_version': wazuh_version,
+        'wazuh_revision': wazuh_revision,
+        'dependencies': dependencies,
+        'targets': targets
+    }
+    yield params
+    logger.info('Restoring Central Component status')
+    GeneralComponentActions.component_restart(params['master'], 'wazuh-manager')
+
+    for indexer_params in params['indexers']:
+        GeneralComponentActions.component_restart(indexer_params, 'wazuh-indexer')
+
+    GeneralComponentActions.component_restart(params['dashboard'], 'wazuh-dashboard')
+    GeneralComponentActions.component_restart(params['master'], 'filebeat')
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_test_environment(wazuh_params):
+    targets = wazuh_params['targets']
+    # Clean the string and split it into key-value pairs
+    targets = targets.replace(' ', '')
+    targets = targets.replace('  ', '')
+    pairs = [pair.strip() for pair in targets.strip('{}').split(',')]
+    targets_dict = dict(pair.split(':') for pair in pairs)
+
+    wazuh_params['master'] = targets_dict.get('wazuh-1')
+    wazuh_params['workers'] = [value for key, value in targets_dict.items() if key.startswith('wazuh-') and key != 'wazuh-1']
+    wazuh_params['indexers'] = [value for key, value in targets_dict.items() if key.startswith('node-')]
+    wazuh_params['dashboard'] = targets_dict.get('dashboard', wazuh_params['master'])
+
+    # If there are no indexers, we choose wazuh-1 by default
+    if not wazuh_params['indexers']:
+        wazuh_params['indexers'].append(wazuh_params['master'])
+
+    wazuh_params['managers'] = {key: value for key, value in targets_dict.items() if key.startswith('wazuh-')}
+
+def test_stop(wazuh_params):
+    GeneralComponentActions.component_stop(wazuh_params['master'], 'wazuh-manager')
+
+    for indexer_params in wazuh_params['indexers']:
+        GeneralComponentActions.component_stop(indexer_params, 'wazuh-indexer')
+
+    GeneralComponentActions.component_stop(wazuh_params['dashboard'], 'wazuh-dashboard')
+    GeneralComponentActions.component_stop(wazuh_params['master'], 'filebeat')
+
+def test_manager_status(wazuh_params):
+    assert 'inactive' in GeneralComponentActions.get_component_status(wazuh_params['master'], 'wazuh-manager'), logger.error(f"The manager in {HostInformation.get_os_name_and_version_from_inventory(wazuh_params['master'])} is not active")
```

Comment:
Wazuh manager

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/5289/commits/7c1d2270fe036e068551866cd700baa9e1b5a2ea